File: src/cpp/01_cmdargs/main.cpp
#include <iostream>

using namespace std;

int main(int argc, char *argv[]) {
  if (argc < 3) {
    cout << "need 2 arguments." << endl;
    return 1;
  }

  auto x = atoi(argv[1]);
  auto y = atoi(argv[2]);
  auto sum = x + y;
  cout << "sum:" << sum << endl;

  return 0;
}

File: src/java/01_cmdargs/Main.java
public class Main {
  public static void main(String... args) {
    if (args.length < 2) {
      System.out.println("need 2 arguments.");
      System.exit(1);
    }

    int x = Integer.parseInt(args[0]);
    int y = Integer.parseInt(args[1]);
    int sum = x + y;

    System.out.println("sum:" + sum);
  }
}

